File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer=1;
    const map=new Map()
    for(const [item, category] of clothes){
        if(map.has(category)) map.set(category, map.get(category)+1)
        else map.set(category, 1)
    }
    const keysArray = Array.from(map.keys());

    for (const key of keysArray) {
        answer *= map.get(key) +1
        console.log(answer)
    }

    return answer-1
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        for(String[] comb:clothes){
            map.put(comb[1], map.getOrDefault(comb[1],0)+1);
        }
        int answer=1;
        Iterator<Integer> iter = map.values().iterator();
        while(iter.hasNext()){
            answer*=(iter.next()+1);
        }
        return answer-1;
    }
}